liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PluginConfigSupport.java

```diff
@@ -382,4 +384,65 @@ private String getPluginConfiguration(MavenProject proj, String pluginGroupId, S
         }
         return null;
     }
+
+    // get loose application configuration file name for project artifact
+    public String getLooseConfigFileName(MavenProject project) {
+        return getPostDeployAppFileName(project) + ".xml";
+    }
+
+    // get loose application file name for project artifact
+    protected String getPostDeployAppFileName(MavenProject project) {
+        return getAppFileName(project, true);
+    }
+
+    // target ear/war produced by war:war, ear:ear, haven't stripped version yet
+    protected String getPreDeployAppFileName(MavenProject project) {
+        return getAppFileName(project, false);
+    }
+
+    protected String getAppFileName(MavenProject project, boolean stripVersionIfConfigured) {
+
+        String name = project.getBuild().getFinalName();
+
+        if (stripVersionIfConfigured && stripVersion) { // TODO stripVersion is set to false when called from dev mojo
+            name = stripVersionFromName(name, project.getVersion());
+        }
+
+        String classifier = MavenProjectUtil.getAppNameClassifier(project);
+        if (classifier != null) {
+            name += "-" + classifier;
+        }
+
+        if (project.getPackaging().equals("liberty-assembly")) {
+            name += ".war";
+        } else if (project.getPackaging().equals("ejb")) {
+            name += ".jar";
+        } else {
+            name += "." + project.getPackaging();
+        }
+
+        return name;
+    }
```

Comment:
On closer look, this PluginConfigSupport class looks like it is for `target/liberty-plugin-config.xml` and while that file does have entries related to looseApplication and applicationFilename, these methods don't look directly related to it.  Perhaps we should consider an intermediate class in the hierarchy between DeployMojoSupport and PluginConfigSupport, maybe "LooseAppSupport" and put these methods in there?  Let me know what you think.

Reply:
I agree that a "LooseAppSupport" file would be best